app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,76 @@
+class ExpensesController < ApplicationController
+  before_action :set_expense, only: %i[show edit update destroy]
+
+  # GET /expenses or /expenses.json
+  def index
+    @category = Category.find(params[:category_id])
+    @expenses = @category.expenses.where(user_id: current_user.id).order('created_at DESC')
+  end
+
+  # GET /expenses/1 or /expenses/1.json
+  def show; end
+
+  # GET /expenses/new
+  def new
+    @expense = Expense.new
+    @category = Category.find(params[:category_id])
+  end
+
+  # GET /expenses/1/edit
+  def edit; end
+
+  # POST /expenses or /expenses.json
+  def create
+    @expense = Expense.new(expense_params)
+    @expense.user = current_user
+    @category = Category.find(params[:category_id])
+
+    respond_to do |format|
+      if @expense.save
+        @expense.categories << @category
```

Comment:
:red_circle: [REQUIRED] I reviewed your current implementation for adding an expense, and I noticed that it does not allow users to add more than one category to an expense. **This is because there is no select field with multiple options or checkboxes to select multiple categories.**

Your current implementation seems to be passing the category through URL parameters

To achieve the requirement of allowing users to associate multiple categories with a single expense, I recommend making the following changes:

1. Modify the form to include a select field with multiple options enabled or checkboxes to select multiple categories.

1. Update the form submission process to handle the selected categories appropriately and store them in the database, as required for a many-to-many association between expenses and categories.

Reply:
Suggested changes by the code reviewer have been implemented. Screen shot attached below for your kind reference.
![image](https://github.com/baqar-abbas/ror-budget-app/assets/117091923/36d61752-e02b-4803-b687-6111c539c85d)